src/sentry/static/sentry/app/views/settings/components/forms/model.tsx

```diff
@@ -283,22 +282,22 @@ class FormModel {
     const endpoint = apiEndpoint || this.options.apiEndpoint || '';
     const method = apiMethod || this.options.apiMethod;
 
-    return new Promise((resolve, reject) => {
-      //should never happen but TS complains if we don't check
-      if (!this.api) {
-        return reject(new Error('Api not set'));
-      }
-      return this.api.request(endpoint, {
+    return new Promise((resolve, reject) =>
+      this.api.request(endpoint, {
         method,
         data,
         success: response => resolve(response),
         error: error => reject(error),
-      });
-    });
+      })
+    );
   }
 
+  /**
+   * Set the value of the form field
+   * if quiet is true, we skip callbacks, validations
+   */
   @action
-  setValue(id: string, value: FieldValue) {
+  setValue(id: string, value: FieldValue, quiet?: boolean) {
```

Comment:
I'd recommend making the 3rd arg an options object so when you're at the callsite, it's more obvious what the 3rd arg is.

```suggestion
  setValue(id: string, value: FieldValue, {quiet}: {quiet?: boolean}) {
```